akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -832,12 +832,17 @@ private[akka] class ActorSystemImpl(
     new Settings(classLoader, config, name, setup)
   }
 
-  val uid: Long = {
-    // to be able to test uid collisions
-    if (settings.config.hasPath("akka.test-only-uid"))
-      settings.config.getLong("akka.test-only-uid")
-    else
-      ThreadLocalRandom.current.nextLong()
+  override def uid: Long = {
```

Comment:
I had this as `lazy val`, but that actually introduces some blocking during the startup phase which seems scary.
It's a `val` in LocalActorRefProvider and a volatile in RemoteActorRefProvider.